spec/rubocop/cop/lint/shadowing_outer_local_variable_spec.rb

```diff
@@ -199,6 +199,34 @@ def some_method
     end
   end
 
+  context 'when a block local variable inside an `if` has same name as an outer scope variable' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        def some_method
+          foo = 1
+
+          if cond
+            bar.each do |foo|
+                         ^^^ Shadowing outer local variable - `foo`.
+            end
+          end
+        end
+      RUBY
+    end
+  end
+
+  context 'when a block local variable has same name as method argument' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        def some_method(foo)
+          1.times do |foo|
+                      ^^^ Shadowing outer local variable - `foo`.
+          end
+        end
+      RUBY
+    end
+  end
```

Comment:
This spec was added to cover [this case](https://github.com/rubocop/rubocop/pull/14036/files#diff-cc952092e003c665fad57bf693892d15804f48a5db8c387627531e963b608c6f) (it's not covered as of now).